### PR TITLE
fix(parser,geometry): the browser pre-pass carries a malformed-record stop to the scanner instead of staying silent

### PR DESCRIPTION
**Pull request by louistrue**
## Summary

When the sharded pre-pass hit an unterminated string or comment, the browser stayed silent: the report #3695 added to the TS scanner never reached the pre-scanned path (#3790).

- `PreScannedEntityIndex.malformedRecordCount` is threaded through `WorkerParser.setEntityIndex`, the `set-entity-index` message (live and queued paths) and `scanIfcEntities`'s pre-scanned branch, so the #3695 diagnostic fires there too. Its `if` is independent of the #3395 "refusal count unreported" diagnostic, so both fire when both apply.
- `ShardColumns.malformedStart` and `StitchedShards.malformedRecordCount` carry the stop out of the stitch with attribution: a stop inside a validated shard counts once and drops later shards; a stop below a shard's validated boundary is an artefact and is not reported. The count is `number | undefined` end to end; absence is "nothing reported", never coerced to 0, so a producer that says nothing cannot read as a clean scan.
- `apps/viewer/src/hooks/entityIndexHandoff.ts` (`forwardEntityIndexTo`) makes the viewer's hop callable and is wired in `useIfcLoader`; the loader's parser is typed as an `EntityIndexSink` and a parameter-tuple check pins the geometry callback's shape.
- The Rust producer does not exist on main yet (#3699 adds the scanner report; its wasm pre-pass side is still to be wired), so `malformedStart` is always absent today. TODOs name #3699 at each arrival point and the changeset says so.

Refs #3790 (closes once #3699's producer is wired through the wasm pre-pass).

## Verification

- RED per hop by reverting the source: parser pre-scanned test 1 failed / 2 passed, shard-stitch 6/6 failed, geometry entity-index 4/4 failed, viewer handoff 3 failed / 2 passed; all GREEN after.
- Mutation-checked: dropping the field from the worker stash, the live postMessage or the queued rebuild each reddens exactly one hop test; restoring the `else if` reddens only the both-apply case; dropping either half of the stitch attribution rule reddens the boundary cases.
- `pnpm typecheck`, `pnpm lint`, parser and geometry suites, `check-module-size` exit 0; viewer suite as 8 shards: 6825 tests, 0 fail.
- Allowlist: two rows raised (geometry-parallel.ts +17, geometry.worker.ts +6; both files sat at zero headroom and the change is spread over five unrelated sites of a single async generator, so no cohesive split exists) and two rows lowered to the measured count.
- Changesets: `@ifc-lite/parser` and `@ifc-lite/geometry` patch.

### CODE REVIEW

File: apps/viewer/src/hooks/useIfcLoader.ts [hunk 1]
```diff
@@ -1607,12 +1609,10 @@ export function useIfcLoader() {
...
-              // null then; the callback no-ops).
-              // #3395: the refusal count travels with the columns, because a
-              // refused record is not IN them.
-              onEntityIndex: (ids, starts, lengths, oversizedIdCount) => {
-                workerParserInstance?.setEntityIndex(ids, starts, lengths, oversizedIdCount);
-              },
+              // null then; the callback no-ops). #3395/#3790: the refusal
+              // count and the malformed-stop flag ride along, because neither
+              // a refused record nor anything after a stop is IN the columns.
+              onEntityIndex: forwardEntityIndexTo(workerParserInstance),
```
[the remark this answers is not in the document]
Comment (louistrue) [reply apps/viewer/src/hooks/useIfcLoader.ts]: Fixed in 570dc51ee: the reviewer was right; the change is in that commit and the test that pins it is in the same file. Resolving.
